Adding UI for VMs and disks

tlitovsk opened the pull request:
Signed-off-by: Tolik Litovsky tlitovsk@redhat.com

Commit:
Adding NFS storage container and volume

Signed-off-by: Tolik Litovsky <tlitovsk@redhat.com>

fabiand commented on specs/nfs-pv.json:
```diff
@@ -0,0 +1,19 @@
+			"path": "/srv/nfs"
+		}
+	}
+}
```
Please add a newline

fabiand commented on specs/nfs-pvc.json:
```diff
@@ -0,0 +1,17 @@
+      }
+    }
+  }
+}
```
Please add a newline

fabiand commented on specs/nfs-server-rc.json:
```diff
@@ -0,0 +1,57 @@
+            }
+        }
+    }
+}
```
Please add a newline

fabiand commented on specs/nfs-server-service.json:
```diff
@@ -0,0 +1,23 @@
+        },
+        "clusterIP": "10.247.126.77"
+    }
+}
```
Please add a newline

fabiand commented on specs/nfs-pv.json:
```diff
@@ -0,0 +1,19 @@
+	},
+	"spec": {
+		"capacity": {
+			"storage": "1Mi"
```
Is this size sufficient for a VM?

fabiand commented on specs/nfs-server-service.json:
```diff
@@ -0,0 +1,23 @@
+        "selector": {
+            "role": "nfs-server"
+        },
+        "clusterIP": "10.247.126.77"
```
I suppose you hard-coded to use  a public IP?
Is this neede?

tlitovsk replied:
clusterIp not external
this the simple way to make connect able by persistantvolume

fabiand commented:
Few comments, but in general no objections. Nice.

Commit:
Starting the volume handler service

Commit:
Storage worker yaml samples

Those yaml files will be used to make the code that lauch the volume worker

Commit:
fixing the creation of disks on kuber storage

Adding debug dockerfiles
Fixing the disk creation chain
Adding status Query for all storage services

Commit:
Alligning all containers to have similar structure

Use python 3
Location of servie and its executio python canonical

Commit:
Adding disks

Added code to add disk
Added kubectl cp emulation

Commit:
switching to templates in the controller code

There are now templates  used to create the kubectl services.
They also now use the sky dns for locating instead of env variables

Commit:
Reworking compute node to generate its own xmls

Commit:
Running VM with mounted disk 

Removing docker file inheritance due to augtool issues
switchin to python to run the vm
addin rest service to acccess the machine
using json to hold vms data

Commit:
Making the usage of libvirtVm class thread safe

removing global libvirtvm object

Commit:
initial display of volumes

Commit:
Containerising the front end

Commit:
Adding delete to qcow manager

Commit:
Working disk list

Todo: connect the delete and add buttons

Commit:
changing put to post to avoid ajax vs bottle issues

the put command get translated in to options
and post works correctly so switching to post

Commit:
Add machine now works

Fixed cpu typo

evol262 commented on container/compute/compute/__main__.py:
```diff
@@ -0,0 +1,101 @@
+from bottle import response
+
+
+def download_vm_defenition():
```
Spelling:

definition

evol262 commented on kubevirtmng/ui/App.jsx:
```diff
@@ -0,0 +1,16 @@
+   render() {
+      return (
+      	 <div>
+			 <DiskTable />
```
Why so many spaces here?

evol262 commented on kubevirtmng/ui/DiskList.jsx:
```diff
@@ -0,0 +1,142 @@
+
+
+class Volume extends React.Component {
+  constructor() {
```
It's better to use:

constructor(props) {
  super(props)
  ...

Just in case Component ever stops setting props itself (after object instantiation). super(props) is a reasonable safeguard, which matches the official docs

evol262 commented on kubevirtmng/ui/DiskList.jsx:
```diff
@@ -0,0 +1,142 @@
+            type: "GET",
+            dataType: 'json',
+            url: 'http://localhost:8083/v1/volumes/'+this.props.data.name+"/",
+            success: function(response){
```
consider indenting a little here for the nested function.

Since you're using babel, you could also use an arrow function, though it's not necessary

evol262 commented on kubevirtmng/ui/DiskList.jsx:
```diff
@@ -0,0 +1,142 @@
+            url: 'http://localhost:8083/v1/volumes/'+this.props.data.name+"/",
+            success: function(response){
+            console.log(response);
+            this.setState({disks: response.disks})
```
.bind(this) gets very trying if you end up with more levels of nesting, which is common, and may happen as your app gets larger.

let self = this

Is a common metapattern (before the functions), then you can use "self.setState({ ...") from any level of anonymous nesting

evol262 commented on kubevirtmng/ui/DiskList.jsx:
```diff
@@ -0,0 +1,142 @@
+              <td>Volume:{this.props.data.name}
+                <table className="datatable table table-striped table-bordered">
+                   <tbody>
+                      {this.state.disks.map((disk, i) => <Disk key = {i} data = {disk} />)}
```
Don't do this as part of return()

Instead, use a variable in render() before return (....

let disks = this.state.disks.map((disk, i) => <Disk key={i} data={disk})

Then in return(), simply use {disks}

evol262 commented on kubevirtmng/ui/DiskList.jsx:
```diff
@@ -0,0 +1,142 @@
+            </tr>
+            <tr>
+              <td>
+                <form className="form-horizontal">
```
Any reason not to use <Form ?

I see react-bootstrap as part of your packages. Might save a lot of boilerplate.

evol262 commented on kubevirtmng/ui/DiskList.jsx:
```diff
@@ -0,0 +1,142 @@
+  render() {
+      return (
+         <tr>
+            <td>Disk:{this.props.data}</td><td>  <button type="button" className="btn btn-primary">Delete</button> </td>
```
Formatting...

multiple elements on one very long line

atom.io is very good about handling this so you don't need to worry about it, but it would be something like

<td>
  Disk: {this.props.data}
</td>
<td>
  <button type=button" className="btn btn-primary">
    Delete
  <button>
</td>

evol262 commented on kubevirtmng/ui/DiskList.jsx:
```diff
@@ -0,0 +1,142 @@
+                </tr>
+              </thead>
+               <tbody>
+                  {this.state.volumes.map((volume, i) => <Volume key = {i} data = {volume} />)}
```
As above -- no "real" logic in return()

evol262 commented on kubevirtmng/ui/DiskList.jsx:
```diff
@@ -0,0 +1,142 @@
+      this.fetchData = this.fetchData.bind(this);
+    };
+
+    fetchData() {
```
This is very similar to fetchData() above.

Have you considered moving this to a helper function not tied to any class, with the uri (and possibly JSON key) to return as parameters, so you can simply:

componentWillMount() {
  this.setState({volumes: fetchData("http://some.uri", volumes)})

evol262 commented on kubevirtmng/ui/index.html:
```diff
@@ -0,0 +1,65 @@
+      <!-- Latest compiled and minified CSS -->
+	    <!-- PatternFly Styles -->
+
+      <!-- jQuery -->
```
In general webpack can suck all of this in for you so you don't need all of these external <script> tags (and you're already using webpack, so these are possibly/probably doing nothing)

Webpack can also pull in css as part of loaders, with:

{
  test: /\.css$/,
  loader: "style-loader!css-loader"
}

(which need to be installed with npm beforehand, but still may be nicer.

Commit:
handling delete requests for disks

Commit:
Making add disk ui work
